src/main/java/org/jenkinsci/plugins/workflow/cps/CpsBodyExecution.java

```diff
@@ -349,7 +361,9 @@ public Next receive(Object o) {
                 c.onFailure(sc, t);
             }
             synchronized (CpsBodyExecution.this) {
-                thread.popContextVariables();
+                if (thread != null) {
+                    thread.popContextVariables();
```

Comment:
As far as I can tell, this call is no longer necessary after the other changes in this PR (in both the success and failure adapters). All of the regression tests I added for JENKINS-53709 pass even if the body of `popContextVariables` is commented out, and the serialized form is identical based on the output of `CpsBodyExecutionTest.popContextVarsOnBodyCompletion`. Perhaps #245 was an incomplete fix for this class of problem, and this PR completes the fix.

Reply:
Perhaps. I will check it. Thanks for the tip.